pkgs/native_toolchain_c/test/cbuilder/cbuilder_cross_windows_host_test.dart

```diff
@@ -36,86 +53,95 @@ void main() {
   });
 
   const dumpbinMachine = {
+    Architecture.arm64: 'ARM64',
     Architecture.ia32: 'x86',
     Architecture.x64: 'x64',
   };
 
   const optimizationLevels = OptimizationLevel.values;
   var selectOptimizationLevel = 0;
+  var selectBuildMode = 0;
 
   final dumpbinFileType = {
     DynamicLoadingBundled(): 'DLL',
     StaticLinking(): 'LIBRARY',
   };
 
-  for (final linkMode in [DynamicLoadingBundled(), StaticLinking()]) {
-    for (final target in targets) {
-      // Cycle through all optimization levels.
-      final optimizationLevel = optimizationLevels[selectOptimizationLevel];
-      selectOptimizationLevel =
-          (selectOptimizationLevel + 1) % optimizationLevels.length;
-      test('CBuilder $linkMode library $target $optimizationLevel', () async {
-        final tempUri = await tempDirForTest();
-        final tempUri2 = await tempDirForTest();
-        final addCUri =
-            packageUri.resolve('test/cbuilder/testfiles/add/src/add.c');
-        const name = 'add';
-
-        final buildInputBuilder = BuildInputBuilder()
-          ..setupShared(
-            packageName: name,
-            packageRoot: tempUri,
-            outputFile: tempUri.resolve('output.json'),
-            outputDirectory: tempUri,
-            outputDirectoryShared: tempUri2,
-          )
-          ..config.setupBuild(
-            linkingEnabled: false,
-            dryRun: false,
-          )
-          ..config.setupShared(buildAssetTypes: [CodeAsset.type])
-          ..config.setupCode(
-            targetOS: OS.windows,
-            targetArchitecture: target,
-            linkModePreference: linkMode == DynamicLoadingBundled()
-                ? LinkModePreference.dynamic
-                : LinkModePreference.static,
-            cCompiler: cCompiler,
+  for (final compiler in compilers.keys) {
+    for (final linkMode in [DynamicLoadingBundled(), StaticLinking()]) {
+      for (final target in targets) {
+        // Cycle through all optimization levels.
+        final optimizationLevel = optimizationLevels[selectOptimizationLevel];
+        selectOptimizationLevel =
+            (selectOptimizationLevel + 1) % optimizationLevels.length;
+        final buildMode = BuildMode.values[selectBuildMode];
+        selectBuildMode = (selectBuildMode + 1) % BuildMode.values.length;
```

Comment:
Is this some kind of fuzzy testing?

Reply:
Heh, it would be fuzz testing if we would randomize the params and run it nightly. Now it's a poor mans combination of trying to cover some combination of params 😄